### PR TITLE
Automatic release to PyPI when tags are released

### PR DESCRIPTION
**Pull request by anshikg**
*Issue #, if available:*

*Description of changes:* This code change allows us to release cloudformation-cli for every tags change

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ammokhov** [comment]
> _Issue #, if available:_
> 
> _Description of changes:_ This code change allows us to release cloudformation-cli for every tags change
> 
> By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

did you test this workflow?

**Comment by anshikg** [comment]
> > _Issue #, if available:_
> > _Description of changes:_ This code change allows us to release cloudformation-cli for every tags change
> > By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
> 
> did you test this workflow?

I am not sure we can till we add tags. We should be able to test this functionality then, Although I am using what linter is using currently so, I don't foresee issues.https://github.com/aws-cloudformation/cfn-python-lint/blob/master/.travis.yml#L34-L45

File: .travis.yml [hunk 1]
```diff
@@ -22,3 +22,15 @@ jobs:
...
+      python: 3.8
+      deploy:
+        provider: pypi
+        user: aws-cloudformation-developers
+        password:
+          secure: ACHz2AGFnR0QWjwRzO3mpO0lnM/WTEQVtY0p8cbf3CrbwJW/DzqAD/IwFWC4dK0CND2VObYuQb0n8mLQKw5k9Bx464NThOu4CEZhRj7oUvMLXAgTAMOsi8GDMMRv7iyBeEvOs5D2cSeLtZEL2sPd3hXcsEG6Mb03DWG66CM3K1/MHmsnKI1dlqgvzzJb2i6dmM2ZAxW90/fAW9Qc4qLLLg93K3vvWm4y+XdOBukf554vPZxRAcEJ4ywjf/e3J1oGDDqs1uWgO4klrZJOjnxaeM+cL1IRCldq1ITQgyczoqBH8eM6QaT0cccs8GiM4z+Le15rwvahm2JV418XbsGrmL39gabLwZarKaVXwlEmLq0LVs0uXoi4RzYfQTULh0vlUprm/4+zhKLEg2YC1fy0dz2V7FB4n2RrC7T72sjbjXtWgTCfw/V+G6DJ2omvTSUsD9oG6L9L0FZ3f0sQV4uvQJ0tzyij+EsbZJ6FFPe71VPCXvTUsFHLN3xRkTSSR2OJg7gO6MmUhvWjewUo+QX9LY6EszU5POZ4Yg4jfeftpJDXUN/dX2pUSr6NuWNvYjf9J5OYvfTtOV6eD4a+2GWLRs7F1pKXVo5K4VCFhIe3qwZh4IF8S5nFUyUVq7WG/CLAh0d6oRIuetKMTzNNrTh1j4vpEVbVofOTDi7BmkIICCo=
+        on:
+          tags: true
+        distributions: sdist bdist_wheel
+        skip_existing: true
```
Comment (PatMyron) [inline .travis.yml:36]: https://docs.travis-ci.com/user/deployment/pypi/
https://docs.travis-ci.com/user/encryption-keys/
https://docs.travis-ci.com/user/build-stages/

https://github.com/aws-cloudformation/cfn-python-lint/blob/eecd57e946ae3ee69067d9f6861467dd9b0a0698/.travis.yml#L6-L8

https://github.com/aws-cloudformation/cfn-python-lint/pull/508

@kddejong is the `stages` section required to make sure the tests pass before deploying? Although I suppose it's safe to assume the tests already passed if it's already been merged:

```
By default, Travis CI will only release from the master branch.
Builds triggered from Pull Requests will never trigger a release.
```

Comment (kddejong) [reply .travis.yml:36]: Yea.  We wanted to validate the tests worked (a final time) before release.  It was also a final check to validate any outside dependencies didn't cause us any issues.  You're right though by the time we do the actual release we have tested the code about 3 times (once on the PR, once when its merged to main, and then again at release).  Obviously this a little redundant.  Your needs may also be dependent on if you are doing branch protection.  We are protecting against direct modifications to the main branch and making sure PRs are tested with an in sync main branch. If you aren't doing those things than you may want to keep those tests.

Comment (anshikg) [reply .travis.yml:36]: I agree with @kddejong I would rather have redundant checks vs miss something. We do have branch protection from direct modification. I would still prefer to keep this change. It is not time sensitive functionality and should not cause any issues by running the tests again.

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,8 @@
 dist: bionic
 language: python
+stages:
+  - Tests
```
Comment (PatMyron) [inline .travis.yml:4]: Think this `Tests` `stage` needs to be used in the `jobs`:
https://github.com/aws-cloudformation/cfn-python-lint/blob/bb1a9c4875a9207da893d1eb68028606b7ff8350/.travis.yml#L11

**Comment by anshikg** [comment]
Please look at this PR: https://github.com/aws-cloudformation/cloudformation-cli/pull/541

**Pull request closed without merging**